compiler/rustc_middle/src/ty/fold.rs

```diff
@@ -849,6 +852,8 @@ struct HasEscapingVarsVisitor {
 }
 
 impl<'tcx> TypeVisitor<'tcx> for HasEscapingVarsVisitor {
+    type BreakTy = FoundEscapingVars;
```

Comment:
This one doesn't even check for the break ty, so keeping `()` and using `is_break()` works fine, too. Or are you trying to protect against the use of `?` with other visitors?

Reply:
https://github.com/rust-lang/rust/pull/78779#discussion_r519021094 would be my reasoning here